apisix/core/request.lua

```diff
@@ -131,26 +128,36 @@ function _M.set_header(ctx, header_name, header_value)
         changed = a6_request.is_request_header_set()
     end
 
-    ngx.req.set_header(header_name, header_value)
+    update_func(header_name, header_value)
 
     if is_apisix_or and not changed then
         -- if the headers are not changed before,
         -- we can only update part of the cache instead of invalidating the whole
         a6_request.clear_request_header()
         if ctx and ctx.headers then
-            ctx.headers[header_name] = header_value
+            if override or not ctx.headers[header_name] then
+                ctx.headers[header_name] = header_value
+            else
+                ctx.headers[header_name] = ctx.headers[header_name] .. header_value
```

Comment:
Do you have test cases to cover this?

Reply:
The Test 15 in `request.t` can cover this.

Reply:
I think it is not corrent.
request headers like this:
```
 Foo: foo
 Foo: bar
 Foo: baz
```
will be a lua table like this:
```
 {"foo", "bar", "baz"}
```

Reply:
Yes, I misunderstood the API of `add_header`, I will fix it later.